ports/esp32/mphalport.c

```diff
@@ -136,7 +137,8 @@ void mp_hal_stdout_tx_strn(const char *str, size_t len) {
     if (release_gil) {
         MP_THREAD_GIL_ENTER();
     }
-    mp_os_dupterm_tx_strn(str, len);
+    mp_os_dupterm_tx_strn(str, ret);
```

Comment:
As per the proposal above, the suggested change for these would be to end up with something equivalent to `ret = MIN(mp_os_dupterm_tx_strn(str, len), ret);`

(Which requires changes to `mp_os_dupterm_tx_strn()` as well.)

Reply:
Those changes to `mp_os_dupterm_tx_strn()` are not trivial. Adding a separate commit for this (we may squash these commits eventually, but I'd like to keep them separate for now).

Reply:
@thingsconnected Sounds good, thanks. Please let us know if it turns into a tarpit of complexity as well. :)